src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
I haven't looked at the code again, just a general comment about @deprecated: the current deprecation notice gives a justification for the deprecation. That's usually not what the caller needs - they just need to know since when this is deprecated, and what to do instead. 

Information about technical debt should probably for formulated as a @todo, in an actionable way. I guess it's ok to give a reason there. But perhaps it's sufficient if the justification is in the commit message.